tests/testthat/test-derive_vars_dt_dtm_utils.R

```diff
@@ -1,32 +1,58 @@
 # convert_date_to_dtm ----
 ## Test 1: Convert a complete -- DTC into a date time object ----
-test_that("convert_date_to_dtm  Test 1: Convert a complete -- DTC into a date time object", {
+test_that("convert_date_to_dtm Test 1: Convert a complete -- DTC into a date time object", {
   expect_equal(
     convert_date_to_dtm("2019-07-18T15:25:52"),
     ymd_hms("2019-07-18T15:25:52")
   )
 })
 
 ## Test 2: Impute incomplete -- DTC into a date time object ----
-test_that("convert_date_to_dtm  Test 2: Impute incomplete -- DTC into a date time object", {
+test_that("convert_date_to_dtm Test 2: Impute incomplete -- DTC into a date time object", {
   expect_equal(
     convert_date_to_dtm("2019-07-18", time_imputation = "23:59:59"),
     ymd_hms("2019-07-18T23:59:59")
   )
 })
 
 ## Test 3: Convert -- DT into a date time object ----
-test_that("convert_date_to_dtm  Test 3: Convert -- DT into a date time object", {
+test_that("convert_date_to_dtm Test 3: Convert -- DT into a date time object", {
   expect_equal(
     convert_date_to_dtm(as.Date("2019-07-18"), time_imputation = "23:59:59"),
     ymd_hms("2019-07-18T23:59:59")
   )
 })
 
-## Test4 4: Keep -- DTM as the original date time object ----
-test_that("convert_date_to_dtm  Test 4: Keep -- DTM as the original date time object", {
+## Test 4: Keep -- DTM as the original date time object ----
+test_that("convert_date_to_dtm Test 4: Keep -- DTM as the original date time object", {
   expect_equal(
     convert_date_to_dtm(ymd_hms("2019-07-18T15:25:52"), time_imputation = "23:59:59"),
     ymd_hms("2019-07-18T15:25:52")
   )
 })
+
+# get_imputation_target_date ----
+## Test 5: get correct target for missing dates ----
+test_that("get_imputation_target_date Test 5: get correct target for missing dates", {
+  expect_equal(
+    get_imputation_target_date("first", NA),
+    list(year = "0000", month = "01", day = "01")
+  )
+})
+
+
+## Test 6: get correct target for missing dates ----
+test_that("get_imputation_target_date Test 6: get correct target for missing dates", {
+  expect_equal(
+    get_imputation_target_date("mid", "04"),
```

Comment:
what is the purpose of the month = "04", isn't it same result with month = NA ?

Reply:
It is not, the second argument is used to for the `preserve` argument, I added one more test to make it a little more clear

Reply:
OK I see it makes a difference. I trust you that it makes sense ;-)